### PR TITLE
Adding irissv: Insertion polishing for SVs - For SV Webinar

### PR DESCRIPTION
**Pull request by NatPRoach**
FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/irissv/irissv.xml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        -->
+        <expand macro="reference"/>
+        <param name="in_vcf" type="data" format="vcf" label="the VCF file with variant calls/supporting reads"/>
+        <param name="in_bam" type="data" format="bam" label="the BAM file containing the reads"/>
+        <!--log_out param doesn't seem to produce output for some reason, commenting out.-->
+        <!-- <param name="output_log" type="boolean" label="Output log file?" checked="true"/> -->
+        <!--
+            Optional arguments
+        -->
+        <param name="padding_before" type="integer" label="the number of bases to output before the variant in REF/ALT fields" value="1" min="1"/>
```
Comment (bgruening) [inline tools/irissv/irissv.xml:44]: can we move the values and min before the label, its easier to see during review.

File: tools/irissv/irissv.xml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        <param name="padding_after" type="integer" label="the number of bases to output after the variant in REF/ALT fields" value="0" min="0"/>
```
Comment (bgruening) [inline tools/irissv/irissv.xml:45]: we could rename "name" here to argument, isn't it?

**Review by bgruening — COMMENTED** [review]
* .shed.yml file missing

**Review by bgruening — COMMENTED** [review]
* the bai file can be removed from this repo
* Is the fai file needed? can it be removed as well?

File: tools/irissv/.shed.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+name: irissv
+owner: iuc
+description: Refine insertion sequences
+homepage_url: https://github.com/galaxyproject/tools-iuc/
+long_description: Implement for Refining Insertion Sequences is a tool which corrects the sequences of structural variant calls (currently only insertions). It uses FalconSense to obtain consensus sequences of the reads surrounding each variant and aligns these sequences back to the reference at the insertion site, resulting in an insertion which takes into account the aggregate information of all supporting reads.
+remote_repository_url: https://github.com/galaxyproject/tools-iuc/
```
Comment (bgruening) [inline tools/irissv/.shed.yml:6]: but the URL to the folder

File: tools/irissv/irissv.xml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        <data name="out_vcf" format="vcf" label="${tool.name} on ${on_string}: Result"/>
+        <!--log_out param doesn't seem to produce output for some reason, commenting out.-->
+        <!-- <data name="out_log" format="txt" label="${tool.name} on ${on_string}: Log">
+            <filter>output_log</filter>
+        </data> -->
+    </outputs>
+    <tests>
+        <!-- #1 default -->
+        <test expect_num_outputs="1">
+            <conditional name="reference_source">
```
Comment (bgruening) [inline tools/irissv/irissv.xml:75]: ideally, you should add a test for the reference from loc file as well, there is some plumping that you need to add to make the all_fasta data table work. Please look at an example tool, most of that can be copied.

Comment (NatPRoach) [reply tools/irissv/irissv.xml:75]: Sure thing. Is there a good example tool to look at?

**Comment by NatPRoach** [comment]
> * the bai file can be removed from this repo
> * Is the fai file needed? can it be removed as well?

fai might not be needed. IIRC it gets produced automatically by irissv if it's not present. I'll remove it in next PR

File: tools/irissv/irissv.xml (hunk 4 above)
Comment (bgruening) [reply tools/irissv/irissv.xml:75]: https://github.com/galaxyproject/tools-iuc/tree/master/tools/exonerate  all those `tool_data_table_conf.xml.*` files are needed.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Great thanks!

**Pull request merged**